lib/m2ts/caption-stream.js

```diff
@@ -801,6 +815,38 @@ Cea608Stream.prototype.isNormalChar = function(char) {
   return (char >= 0x20 && char <= 0x7f);
 };
 
+/**
+ * Configures roll-up
+ *
+ * @param  {Integer} pts         Current PTS
+ * @param  {Integer} newBaseRow  Used by PACs to slide the current window to
+ *                               a new position
+ */
+Cea608Stream.prototype.setRollUp = function(pts, newBaseRow) {
+  // Reset the base row to the bottom row when switching modes
+  if (this.mode_ !== 'rollUp') {
+    this.row_ = BOTTOM_ROW;
+    this.mode_ = 'rollUp';
+    // Spec says to wipe memories when switching to roll-up
+    this.flushDisplayed(pts);
+    this.nonDisplayed_ = createDisplayBuffer();
+    this.displayed_ = createDisplayBuffer();
+  }
+
+  if (newBaseRow !== undefined && newBaseRow !== this.row_) {
+    // slide the window around if we need to
+    for (var i = 0; i < this.rollUpRows_; i++) {
+      this.displayed_[newBaseRow - i] = this.displayed_[this.row_ - i];
+      this.displayed_[this.row_ - i] = '';
+    }
+  }
+
+  if (newBaseRow === undefined) {
+    newBaseRow = this.row_;
+  }
+  this.topRow_ = newBaseRow - this.rollUpRows_ + 1;
```

Comment:
Correct. `this.row_` is set [here](https://github.com/videojs/mux.js/pull/171/files#diff-7de0c0c8257aefbda729beef42a039e4R579) since we can have row-changing PACs in modes other than roll-up.